internal/controllers/topology/cluster/conditions_test.go

```diff
@@ -95,6 +98,69 @@ func TestReconcileTopologyReconciledCondition(t *testing.T) {
 				".status.observedGeneration == .metadata.generation is true. If this is not the case either ClusterClass reconciliation failed or the ClusterClass is paused",
 			wantErr: false,
 		},
+		{
+			name:         "should set the condition to false if the there is a blocking annotation hook",
+			reconcileErr: nil,
+			cluster: &clusterv1.Cluster{
+				ObjectMeta: metav1.ObjectMeta{
+					Annotations: map[string]string{
+						clusterv1.BeforeClusterUpgradeHookAnnotationPrefix + "/test": "true",
+					},
+				},
+			},
+			s: &scope.Scope{
+				HookResponseTracker: func() *scope.HookResponseTracker {
+					hrt := scope.NewHookResponseTracker()
+					hrt.Add(runtimehooksv1.BeforeClusterUpgrade, &runtimehooksv1.BeforeClusterUpgradeResponse{
+						CommonRetryResponse: runtimehooksv1.CommonRetryResponse{
+							CommonResponse: runtimehooksv1.CommonResponse{
+								Message: fmt.Sprintf("annotation [%s] is set", clusterv1.BeforeClusterUpgradeHookAnnotationPrefix+"/test"),
+							},
+							RetryAfterSeconds: int32(20 * 60),
```

Comment:
```suggestion
							RetryAfterSeconds: 20 * 60,
```
nit, same below (I was just surprised when comparing with the prod code)